packages/cloudflare/src/api/cloudflare-context.ts

```diff
@@ -53,26 +61,48 @@ type InternalGlobalThis<
   CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
   Context = ExecutionContext,
->(): CloudflareContext<CfProperties, Context> {
+>(options: { async: true }): Promise<CloudflareContext<CfProperties, Context>>;
+export function getCloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+>(options?: { async: false }): CloudflareContext<CfProperties, Context>;
+export function getCloudflareContext<
+  CfProperties extends Record<string, unknown> = IncomingRequestCfProperties,
+  Context = ExecutionContext,
+>(
+  options: GetCloudflareContextOptions = { async: false }
+): CloudflareContext<CfProperties, Context> | Promise<CloudflareContext<CfProperties, Context>> {
   const global = globalThis as InternalGlobalThis<CfProperties, Context>;
 
   const cloudflareContext = global[cloudflareContextSymbol];
 
+  // whether `getCloudflareContext` is run in "async mode"
+  const asyncMode = options.async;
+
   if (!cloudflareContext) {
-    // For SSG Next.js creates (jest) workers that run in parallel, those don't get the current global
-    // state so they can't get access to the cloudflare context, unfortunately there isn't anything we
-    // can do about this, so the only solution is to error asking the developer to opt-out of SSG
-    // Next.js sets globalThis.__NEXT_DATA__.nextExport to true for the worker, so we can use that to detect
-    // that the route is being SSG'd (source: https://github.com/vercel/next.js/blob/4e394608423/packages/next/src/export/worker.ts#L55-L57)
-    if (global.__NEXT_DATA__?.nextExport === true) {
+    // The sync mode of `getCloudflareContext`, relies on the context being set on the global state
+    // by either the worker entrypoint (in prod) or by `initOpenNextCloudflareForDev` (in dev), neither
+    // can work during SSG since for SSG Next.js creates (jest) workers that don't get access to the
+    // normal global state so we throw with a helpful error message.
+    // Note: Next.js sets globalThis.__NEXT_DATA__.nextExport to true for SSG routes
+    // source: https://github.com/vercel/next.js/blob/4e394608423/packages/next/src/export/worker.ts#L55-L57)
+    if (!asyncMode && global.__NEXT_DATA__?.nextExport === true) {
       throw new Error(
         `\n\nERROR: \`getCloudflareContext\` has been called in a static route` +
-          ` that is not allowed, please either avoid calling \`getCloudflareContext\`` +
-          ` in the route or make the route non static (for example by exporting the` +
-          ` \`dynamic\` route segment config set to \`'force-dynamic'\`.\n`
+          ` that is not allowed, this can be solved in different ways:\n\n` +
+          ` - call \`getCloudflareContext({async: true})\` to use the \`async\` mode\n` +
+          ` - avoid calling \`getCloudflareContext\` in the route\n` +
+          ` - make the route non static\n`
       );
     }
 
+    if (options.async) {
```

Comment:
Maybe 
- you can move this if as the first nested if and you can drop !asyncMode l. 89.
- be consistent in using options.async or asyncMode
- do a release once this gets merged

Reply:
> be consistent in using options.async or asyncMode

oops, not sure how I missed using `asyncMode` here when I refactored this, my bad, thanks for noticing it 👍